src/SonarQube.Client.Tests/SonarQubeService_GetRuleByKeyAsync.cs

```diff
@@ -0,0 +1,181 @@
+        ""createdAt"": ""2019-01-10T14:00:14+0100"",
+        ""updatedAt"": ""2019-01-10T14:00:14+0100""
+      }
+    ],
+    ""csharpsquid:S4524"": [
+      {
+        ""qProfile"": ""quality-profile-1"",
+        ""inherit"": ""NONE"",
+        ""severity"": ""CRITICAL"",
+        ""params"": [],
+        ""createdAt"": ""2019-01-10T14:00:14+0100"",
+        ""updatedAt"": ""2019-01-10T14:00:14+0100""
+      }
+    ],
+    ""csharpsquid:S2342"": [
+      {
+        ""qProfile"": ""quality-profile-1"",
+        ""inherit"": ""NONE"",
+        ""severity"": ""MINOR"",
+        ""params"": [
+          {
+            ""key"": ""format"",
+            ""value"": ""^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$""
+          },
+          {
+            ""key"": ""flagsAttributeFormat"",
+            ""value"": ""^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?s$""
+          }
+        ],
+        ""createdAt"": ""2019-01-10T14:00:14+0100"",
+        ""updatedAt"": ""2019-01-10T14:00:14+0100""
+      }
+    ]
+  },
+  ""qProfiles"": {
+    ""quality-profile-1"": {
+      ""name"": ""Sonar way"",
+      ""lang"": ""cs"",
+      ""langName"": ""C#""
+    }
+  }
+}
+");
+
+            var result = await service.GetRuleByKeyAsync("csharpsquid:S2342", CancellationToken.None);
+
+            messageHandler.VerifyAll();
+
+            result.Should().NotBeNull();
+
+            result.Key.Should().Be("S2342");
+            result.RepositoryKey.Should().Be("csharpsquid");
+            result.Description.Should().Be("Html Description");
+
+            result.DescriptionSections.Count.Should().Be(2);
+            result.DescriptionSections[0].Key.Should().Be("key1");
+            result.DescriptionSections[0].HtmlContent.Should().Be("content1");
+            result.DescriptionSections[0].Context.Should().BeNull();
+
+            result.DescriptionSections[1].Key.Should().Be("key2");
+            result.DescriptionSections[1].HtmlContent.Should().Be("content2");
+            result.DescriptionSections[1].Context.Should().NotBeNull();
+            result.DescriptionSections[1].Context.Key.Should().Be("key");
+            result.DescriptionSections[1].Context.DisplayName.Should().Be("displayName");
+        }
+
+        [TestMethod]
+        public async Task GetRuleByKeyAsync_RuleIsNotFound_ReturnsNull()
+        {
+            await ConnectToSonarQube();
+
+            SetupRequest("api/rules/search?rule_key=csharpsquid%3AS2342XX&f=repo%2CinternalKey%2Cparams%2Cactives%2ChtmlDesc%2CdescriptionSections%2CeducationPrinciples&p=1&ps=500", @"{
+    ""total"": 0,
+    ""p"": 1,
+    ""ps"": 100,
+    ""rules"": [],
+    ""actives"": {},
+    ""qProfiles"": {},
+    ""paging"": {
+        ""pageIndex"": 1,
+        ""pageSize"": 100,
+        ""total"": 0
+    }
+}
+");
+
+            var result = await service.GetRuleByKeyAsync("csharpsquid:S2342XX", CancellationToken.None);
+
+            messageHandler.VerifyAll();
+
+            result.Should().BeNull();
+        }
```

Comment:
All of the existing API methods have a test that checks what happens if the server is not connected e.g. `GetRulesAsync_NotConnected`